### PR TITLE
fix JUnit report format

### LINKED ISSUE #273: Junit output does not seem valid (parsing error in Jenkins)
Hello there 👋 

First of all thanks for this really useful tool!

While trying the junit output feature from the latest `v1.10.0`, I was not able to feed it to Jenkins as a Junit standard report.
The error happens during the Junit parsing:
```
java.lang.NullPointerException
	at hudson.tasks.junit.CaseResult.getPackageName(CaseResult.java:399)
	at hudson.tasks.junit.TestResult.tally(TestResult.java:795)
	at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:145)
	at hudson.FilePath.act(FilePath.java:1161)
	at hudson.FilePath.act(FilePath.java:1144)
	at hudson.tasks.junit.JUnitParser.parseResult(JUnitParser.java:107)
	at hudson.tasks.junit.JUnitResultArchiver.parse(JUnitResultArchiver.java:149)
	at hudson.tasks.junit.JUnitResultArchiver.parseAndSummarize(JUnitResultArchiver.java:243)
	at hudson.tasks.junit.pipeline.JUnitResultsStepExecution.run(JUnitResultsStepExecution.java:63)
	at hudson.tasks.junit.pipeline.JUnitResultsStepExecution.run(JUnitResultsStepExecution.java:29)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

It looks like that the XML generated by CST. is not valid:

```
<testsuites failures="0" tests="14" time="0"><testsuite><testcase name="File Content Test: image user created as non-root user" time="0"></testcase><testcase name="File Content Test: subuid contains user with 65,536 subordinate UIDs" time="0"></testcase><testcase name="File Content Test: subgid contains user 65,536 subordinate GIDs" time="0"></testcase><testcase name="File Existence Test: Google Container Test CLI" time="0"></testcase><testcase name="File Existence Test: HADOLINT" time="0"></testcase><testcase name="File Existence Test: Root" time="0"></testcase><testcase name="File Existence Test: img" time="0"></testcase><testcase name="File Existence Test: img newuidmap" time="0"></testcase><testcase name="File Existence Test: img newgidmap" time="0"></testcase><testcase name="File Existence Test: subuid" time="0"></testcase><testcase name="File Existence Test: subgid" time="0"></testcase><testcase name="File Existence Test: user directory" time="0"></testcase><testcase name="File Existence Test: home user" time="0"></testcase><testcase name="Metadata Test" time="0"></testcase></testsuite></testsuites>
```
(The `cst.yml` file is public and available here: https://github.com/jenkins-infra/docker-builder/blob/main/cst.yml ).

While iterating from an existing working Junit report from one of my Java projects, I was able to correctly feed this modified XML report to Jenkins:

```
<?xml version="1.0" encoding="UTF-8"?>
<testsuite xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
  xsi:noNamespaceSchemaLocation="https://maven.apache.org/surefire/maven-surefire-plugin/xsd/surefire-test-report-3.0.xsd" 
  version="3.0" name="report" failures="0" tests="14" time="0" errors="0" skipped="0">
  <testcase name="File Content Test: image user created as non-root user" time="0"></testcase>
  <testcase name="File Content Test: subuid contains user with 65,536 subordinate UIDs" time="0"></testcase>
  <testcase name="File Content Test: subgid contains user 65,536 subordinate GIDs" time="0"></testcase>
  <testcase name="File Existence Test: Google Container Test CLI" time="0"></testcase>
  <testcase name="File Existence Test: HADOLINT" time="0"></testcase>
  <testcase name="File Existence Test: Root" time="0"></testcase>
  <testcase name="File Existence Test: img" time="0"></testcase>
  <testcase name="File Existence Test: img newuidmap" time="0"></testcase>
  <testcase name="File Existence Test: img newgidmap" time="0"></testcase>
  <testcase name="File Existence Test: subuid" time="0"></testcase>
  <testcase name="File Existence Test: subgid" time="0"></testcase>
  <testcase name="File Existence Test: user directory" time="0"></testcase>
  <testcase name="File Existence Test: home user" time="0"></testcase>
  <testcase name="Metadata Test" time="0"></testcase>
</testsuite>
```

It includes the following changes (which might be a cause or not, I did not dig enough in the details so we might be able to remove some of these changes and still have a valid XML report):

* Adding an `<?xml version="1.0" encoding="UTF-8"?>` first line
* Switching from the singular form `testcases` to a singular form `testcase`
* Adding a DTD grammar to the `testcase` markup
* Following the grammar validation, adding 2 missing attributes to the `testcase` markup: `errors` and `skipped`

I'm not really sure where to search for the correct way of validating a Junit XML report, and I did not tried (yet) another tool (as Gitlab or Github Action, or even a Maven surefire report) to parse the Junit report, but I wanted to report this issue to start the discussion.

If you want to reproduce the "Jenkins" case I have, you can use https://github.com/dduportal/jenkins-k8s (which is public) to spawn a Jenkins instance with all the necessary plugins and setup. At the time of writing this comment, I was using the commit https://github.com/dduportal/jenkins-k8s/tree/74ddada3b24ec34fec20c5810a9496443087f3f7 which means Jenkins `2.263.2` and Junit's Jenkins plugin `1.48`.

### PR DESCRIPTION
**Pull request by coopernetes**
Fixes #273 & #304

I added two new types to handle the JUnit specific marshaling. This probably can be implemented a bit better rather than inline structs + new types just for making encoding/xml happy.

tests pass. Had to make a tweak to tests/structure_test_tests.sh get the script to run correctly. Could use a pair of eyes to make sure its kosher 👀 
```bash
$ make test
GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -ldflags " -X github.com/GoogleContainerTools/container-structure-test/pkg/version.version=v1.11.0 -X github.com/GoogleContainerTools/container-structure-test/pkg/version.buildDate=2022-09-10T23:12:32Z " -o out/container-structure-test-linux-amd64 github.com/GoogleContainerTools/container-structure-test/cmd/container-structure-test
cp ./out/container-structure-test-linux-amd64 out/container-structure-test
./tests/structure_test_tests.sh
##
# Build the newest 'container structure test' binary
##
make[1]: Entering directory '/home/tom/oss/container-structure-test'
make[1]: 'out/container-structure-test' is up to date.
make[1]: Leaving directory '/home/tom/oss/container-structure-test'
make[1]: Entering directory '/home/tom/oss/container-structure-test'
shasum -a 256 out/container-structure-test-linux-amd64 &> out/container-structure-test-linux-amd64.sha256
GOOS=darwin GOARCH=amd64 CGO_ENABLED=0 go build -ldflags " -X github.com/GoogleContainerTools/container-structure-test/pkg/version.version=v1.11.0 -X github.com/GoogleContainerTools/container-structure-test/pkg/version.buildDate=2022-09-10T23:12:33Z " -o out/container-structure-test-darwin-amd64 github.com/GoogleContainerTools/container-structure-test/cmd/container-structure-test
1af7e99e80ab87be17e42c2215775213b00bf26b94d912e841e13074c2aa1e28  out/container-structure-test-linux-amd64
shasum -a 256 out/container-structure-test-darwin-amd64 &> out/container-structure-test-darwin-amd64.sha256
GOOS=windows GOARCH=amd64 CGO_ENABLED=0 go build -ldflags " -X github.com/GoogleContainerTools/container-structure-test/pkg/version.version=v1.11.0 -X github.com/GoogleContainerTools/container-structure-test/pkg/version.buildDate=2022-09-10T23:12:33Z " -o out/container-structure-test-windows-amd64 github.com/GoogleContainerTools/container-structure-test/cmd/container-structure-test
cb459a834dd408c3c49871996de887ef4869e1b3a39969ddbc5bbb40ea4eb5e3  out/container-structure-test-darwin-amd64
cp out/container-structure-test-windows-amd64 out/container-structure-test-windows-amd64.exe
shasum -a 256 out/container-structure-test-windows-amd64.exe &> out/container-structure-test-windows-amd64.exe.sha256
rm out/container-structure-test-windows-amd64
make[1]: Leaving directory '/home/tom/oss/container-structure-test'
make[1]: Entering directory '/home/tom/oss/container-structure-test'
docker build -t gcr.io/gcp-runtimes/container-structure-test:latest .
b52f3be912ee64f74d5ea8b976a1810c44d361b050ebdbcc30192549343fbd31  out/container-structure-test-windows-amd64.exe
Sending build context to Docker daemon   75.7MB
Step 1/3 : FROM ubuntu:20.04
 ---> a0ce5a295b63
Step 2/3 : ADD out/container-structure-test /container-structure-test
 ---> 0baf39323714
Step 3/3 : ENTRYPOINT ["/container-structure-test"]
 ---> Running in 45c8c2fdfbe1
Removing intermediate container 45c8c2fdfbe1
 ---> d6e10d9b923c
Successfully built d6e10d9b923c
Successfully tagged gcr.io/gcp-runtimes/container-structure-test:latest
make[1]: Leaving directory '/home/tom/oss/container-structure-test'
##
# Positive Test Case
##
PASS: Success test case passed
##
# Metadata Test Case
##
PASS: Metadata success test case for docker driver
PASS: Metadata success test case for tar driver
PASS: Metadata success test case for host driver
##
# Failure Test Case
##
FATA[0000] FAIL                                         
PASS: Failure test failed
###
# OCI layout test case
###
PASS: oci failing test case
PASS: oci success test case
```

### CODE REVIEW

**Review by loosebazooka — APPROVED** [review]

**Comment by loosebazooka** [comment]
I'm not actually sure what this integration test is waiting for. If you want to trigger this with some sort of rebase or other minor change, maybe we can merge it in.

**Comment by coopernetes** [comment]
> 1 workflow awaiting approval
> First-time contributors need a maintainer to approve running workflows. [Learn more.](https://docs.github.com/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks)
Looks like it's waiting on a manual approval. @loosebazooka would you mind kicking it off?

**Comment by coopernetes** [comment]
@nkubala or @dlorenc , would either of you folks be able to approve the workflow & run the tests (should be passing now)?

**Comment by loosebazooka** [comment]
Yeah I dunno what's going on. But I approved it now

**Pull request merged**
